pkg/destroy/powervs/power-subnet.go

```diff
@@ -0,0 +1,155 @@
+package powervs
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const powerSubnetTypeName = "powerSubnet"
+
+// listPowerSubnets lists subnets in the Power Server.
+func (o *ClusterUninstaller) listPowerSubnets() (cloudResources, error) {
+	o.Logger.Debugf("Listing Power Server Subnets")
+
+	if o.instanceClient == nil {
+		o.Logger.Infof("Skipping deleting Power service subnets because no service instance was found")
+		result := []cloudResource{}
+		return cloudResources{}.insert(result...), nil
+	}
+
+	networks, err := o.networkClient.GetAll()
+	if err != nil {
+		o.Logger.Warnf("Error networkClient.GetAll: %v", err)
+		return nil, err
+	}
+
+	var foundOne = false
+
+	result := []cloudResource{}
+	for _, network := range networks.Networks {
+		if strings.Contains(*network.Name, o.InfraID) {
+			foundOne = true
+			o.Logger.Debugf("listPowerSubnets: FOUND: %s, %s", *network.NetworkID, *network.Name)
+			result = append(result, cloudResource{
+				key:      *network.NetworkID,
+				name:     *network.Name,
+				status:   "",
+				typeName: powerSubnetTypeName,
+				id:       *network.NetworkID,
+			})
+		}
+	}
+	if !foundOne {
```

Comment:
Why do you need this var? Can't you just check: 
```suggestion
	if len(result) == 0 {
```

Reply:
ok